src/controllers/admin.js

```diff
@@ -1,125 +1,154 @@
 const mongoose = require('mongoose');
 const Admin = require('../models/Admin');
 
-const createAdmin = (req, res) => {
+const createAdmin = async (req, res) => {
   const {
     firstName, lastName, dni, phone, email, city, password,
   } = req.body;
-  Admin.findOne({ dni })
-    .then((found) => {
-      if (found) {
-        return res.status(404).json({
-          message: `An admin with DNI ${dni} already exists`,
-          data: undefined,
-          error: true,
-        });
-      }
-      return Admin.create({
-        firstName, lastName, dni, phone, email, city, password,
+  try {
+    const found = await Admin.findOne({ dni });
+    if (found) {
+      return res.status(404).json({
+        message: `An admin with DNI ${dni} already exists`,
+        data: undefined,
+        error: true,
       });
-    })
-    .then((result) => res.status(201).json({
+    }
+    const result = await Admin.create({
+      firstName, lastName, dni, phone, email, city, password,
+    });
+    return res.status(201).json({
       message: 'New admin created',
       data: result,
       error: false,
-    }))
-    .catch((error) => res.status(400).json({
-      message: 'An error ocurred!',
-      error,
-    }));
+    });
+  } catch (error) {
+    return res.status(400).json({
+      message: error.message,
+      data: undefined,
+      error: true,
+    });
+  }
 };
 
 const updateAdmin = async (req, res) => {
   const { id } = req.params;
   const {
     firstName, lastName, dni, phone, email, city, password,
   } = req.body;
-  if (!mongoose.Types.ObjectId.isValid(id)) {
-    return res.status(404).json({
-      message: 'Id is not a valid one',
-      data: undefined,
-      error: true,
+
+  try {
+    if (!mongoose.Types.ObjectId.isValid(id)) {
+      return res.status(404).json({
+        message: 'Id is not a valid one',
+        data: undefined,
+        error: true,
+      });
+    }
+
+    const actualAdmin = await Admin.findById(id);
+    const adminProperties = Object.keys(actualAdmin.toObject()).slice(1, -1);
+    let changes = false;
+    adminProperties.forEach((property) => {
+      if (req.body[property]
+      && req.body[property].toString() !== actualAdmin[property].toString()) {
+        changes = true;
+      }
     });
-  }
-  const actualAdmin = await Admin.findById(id);
-  const adminProperties = Object.keys(actualAdmin.toObject()).slice(1, -1);
-  let changes = false;
-  adminProperties.forEach((property) => {
-    if (req.body[property]
-    && req.body[property].toString() !== actualAdmin[property].toString()) {
-      changes = true;
+    if (!changes) {
+      return res.status(200).json({
+        message: 'There were no changes',
+        data: undefined,
+        error: false,
+      });
     }
-  });
-  if (!changes) {
-    return res.status(200).json({
-      message: 'There were no changes',
-      data: undefined,
+    const found = await Admin.findOne({ dni });
```

Comment:
This is great! I hadn't think of validating this in the update too, it's a great validation if we don't want to have repeating entities, you should post this on slack! Great Idea! 🤩 (I'll do this on my code too)

Reply:
Yup that was an important part that was missing!